### PR TITLE
Add `XButton` + fix various buttons on Mac Catalyst

### LINKED ISSUE #401: Mac Catalyst UI issues
### Bookmarks

- [ ] These should appear as simple list items, instead of button-looking items. #549 

Mac Catalyst, macOS Ventura v13.3.1

image

### Floor Filter

- [ ] Button styles

image

Also, it doesn't appear as if the `ProgressView` is showing when the map is loading.

### Utility Network Trace Tool

- [ ] Trace Results combo box is full-width; it should have some padding.

image

### PR DESCRIPTION
**Pull request by dfeinzimer**
Closes #401 & #857

### CODE REVIEW

File: Sources/ArcGISToolkit/Components/FloorFilter/LevelSelector.swift [hunk 1]
```diff
@@ -70,8 +70,8 @@ extension LevelSelector {
     /// The system name of the icon that reflects the current state of `isCollapsed`.
     var iconForCollapsedState: String {
         switch (isCollapsed, isTopAligned) {
-        case (true, true), (false, false): return "chevron.down.circle"
-        case (true, false), (false, true): return "chevron.up.circle"
+        case (true, true), (false, false): return "chevron.down"
+        case (true, false), (false, true): return "chevron.up"
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/FloorFilter/LevelSelector.swift:74]: Changed to match the expand/collapse symbol in SearchView.

**Comment by dfeinzimer** [comment]
Before/After diffs

[FormElements.zip](https://github.com/user-attachments/files/17927085/FormElements.zip)
[CalibrationView.zip](https://github.com/user-attachments/files/17927087/CalibrationView.zip)
[Archive.zip](https://github.com/user-attachments/files/17927093/Archive.zip)
[Search.zip](https://github.com/user-attachments/files/17927095/Search.zip)

File: Sources/ArcGISToolkit/Components/FloorFilter/LevelSelector.swift [hunk 2]
```diff
@@ -85,7 +85,9 @@ extension LevelSelector {
         } label: {
             Image(systemName: iconForCollapsedState)
                 .padding(.toolkitDefault)
+                .contentShape(Rectangle())
```
Comment (mhdostal) [inline Sources/ArcGISToolkit/Components/FloorFilter/LevelSelector.swift:88]: What does this do for the image?

**Review by mhdostal — APPROVED** [review]
Just the one question, for my own information.

File: Sources/ArcGISToolkit/Common/XButton.swift [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        case .clear:
+            String.clear
+        case .dismiss:
+            String.dismiss
+        }
+    }
+}
+
+@available(iOS 17.0, *)
+#Preview {
```
Comment (zkline101) [inline Sources/ArcGISToolkit/Common/XButton.swift:66]: Should this preview be this complex? Is that a common thing? I guess I was expecting it to be very simple where just the XButton is being displayed.

**Review by zkline101 — APPROVED** [review]
Code looks good and I tested on Vision Pro device and it looks good.

File: Sources/ArcGISToolkit/Common/XButton.swift (hunk 3 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Common/XButton.swift:66]: Took some things out to simplify it a bit.

Originally I had planned a second preview to demonstrate it as a clear button. Thoughts?

```swift
@available(iOS 17.0, *)
#Preview("Clear Content") {
    @Previewable @State var text = "Hello, World!"
    HStack {
        TextField(text: $text) { }
        XButton(.clear) {
            text.removeAll()
        }
    }
    .padding()
}
```

Comment (zkline101) [reply Sources/ArcGISToolkit/Common/XButton.swift:66]: I'm wondering if we even need the sheet. But anyway this isn't a huge deal I was just curious about what is expected in a preview but I don't think that is clear.

**Review by zkline101 — APPROVED** [review]

**Review by mhdostal — APPROVED** [review]

**Pull request merged**